tests/components/tomorrowio/test_init.py

```diff
@@ -47,6 +49,41 @@ async def test_load_and_unload(hass: HomeAssistant) -> None:
     assert len(hass.states.async_entity_ids(WEATHER_DOMAIN)) == 0
 
 
+async def test_update_intervals(hass: HomeAssistant) -> None:
+    """Test coordinator update intervals."""
+    data = _get_config_schema(hass, SOURCE_USER)(MIN_CONFIG)
+    data[CONF_NAME] = "test"
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=data,
+        options={CONF_TIMESTEP: 1},
+        unique_id=_get_unique_id(hass, data),
+        version=1,
+    )
+    config_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert hass.data[DOMAIN][
```

Comment:
Please move time forward instead and assert that the appropriate number of api calls are made for every case.

https://github.com/home-assistant/core/blob/371dfd85c8b61a62c48526778b377c090e6c0982/tests/common.py#L375-L379